### ISSUE TITLE
kernal panics when virtualbox installed

### ISSUE THREAD

**Issue by allowthere** [NixOS/nixpkgs#54450]
## Issue description

Kernel panics before step 2 of boot

### Steps to reproduce

When I add `virtualisation.virtualbox.host.enable=true` to my /etc/nixos/configuration.nix, I have a kernel panic on boot.

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.16, NixOS, 18.09.1974.50f41ea2fcf (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(root): `"nixos-18.09.1974.50f41ea2fcf"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by teto** [comment]
could help to see what's the kernel panic like/kernel configuration.

**Comment by eadwu** [comment]
Using 4.20 is fine, but you need to have this #53120 somewhere since virtualbox 5.x doesn't support 4.20.

**Comment by allowthere** [comment]
Sorry. It was my error.

What happened was that when I _first_ installed up nixos, I didn't mount /boot when running nixos-generate-config. I mounted it later in the install process, but it was never documented in hardware-configuration.nix. 

So when I booted into nixos, everything worked well. Except when I restarted after a `nixos-rebuild switch`, I would get a kernel panic. For some reason, sometimes the kernel panic was longer than my screen, so I couldn't get what caused it, so I thought it was virtalbox's fault (since that's the only thing I installed post-reboot). It turns out that nixos-rebuild wrote to /boot on the main partition (ZFS in my case), and on reboot it couldn't an init.

So I remounted it, switched to grub (from systemd), and it works fine now.

Sorry.
